### PR TITLE
NPC Health Bug fix - Maybe

### PR DESCRIPTION
**Pull request by Undid-Iridium**
Not my area of expertise, nor did I write the original code so guessing - if better solution exists, take it.

## Description
**Describe the changes** 
NPC hp bug

**What is the current behavior?** (You can also link to an open issue here)
NPC hp doesn't use exiled health stat

**What is the new behavior?** (if this is a feature change)
NPC hp is now using exiled health stat.

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Maybe

**Other information**:
https://discord.com/channels/656673194693885975/1300857604620161064

<br />

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentations
<br />

## Submission checklist

- [X] I have checked the project can be compiled
- [?] I have tested my changes and it worked as expected

### Patches (if there are any changes related to Harmony patches)
- [X] I have checked no IL patching errors in the console

### Other
- [X] Still requires more testing

### CODE REVIEW

**Review by Misfiy — CHANGES_REQUESTED** [review]
Feels like Npc class should fix it (it worked a while ago, no?)

**Comment by Undid-Iridium** [comment]
> Feels like Npc class should fix it (it worked a while ago, no?)

?

**Comment by VALERA771** [comment]
Bug seems to occur only on npcs so this changes should be made to npc class, not player

File: EXILED/Exiled.API/Features/Npc.cs [hunk 1]
```diff
@@ -131,6 +131,8 @@ public override Vector3 Position
         /// <param name="conn">The NetworkConnection to retrieve the NPC for.</param>
         /// <returns>The NPC associated with the NetworkConnection, or <c>null</c> if not found.</returns>
         public static new Npc? Get(NetworkConnection conn) => Player.Get(conn) as Npc;
+        
+
```
Comment (louis1706) [inline EXILED/Exiled.API/Features/Npc.cs:135]:
```suggestion
```

File: EXILED/Exiled.API/Features/Npc.cs [hunk 2]
```diff
@@ -273,6 +276,15 @@ public static Npc Spawn(string name, RoleTypeId role = RoleTypeId.None, bool ign
 
                 if (position is not null)
                     npc.Position = position.Value;
+
+                if (PlayerMaxHealth != float.NegativeInfinity)
+                {
+                    npc.MaxHealth = PlayerMaxHealth;
+                }
```
Comment (louis1706) [inline EXILED/Exiled.API/Features/Npc.cs:283]:
```suggestion
                }

```

File: EXILED/Exiled.API/Features/Npc.cs [hunk 3]
```diff
@@ -217,7 +219,8 @@ public static Npc Spawn(string name, RoleTypeId role, int id = 0, string userId
         /// <param name="userId">The userID of the NPC for authentication. Defaults to the Dedicated ID.</param>
         /// <param name="position">The position where the NPC should spawn. If null, the default spawn location is used.</param>
         /// <returns>The <see cref="Npc"/> spawned.</returns>
-        public static Npc Spawn(string name, RoleTypeId role = RoleTypeId.None, bool ignored = false, string userId = PlayerAuthenticationManager.DedicatedId, Vector3? position = null)
+        public static Npc Spawn(string name, RoleTypeId role = RoleTypeId.None, bool ignored = false,
+            string userId = PlayerAuthenticationManager.DedicatedId, Vector3? position = null, float PlayerHealth = float.NegativeInfinity, float PlayerMaxHealth = float.NegativeInfinity)
```
Comment (louis1706) [inline EXILED/Exiled.API/Features/Npc.cs:223]: Fix error

**Review by louis1706 — CHANGES_REQUESTED** [review]

**Comment by louis1706** [comment]
NPC work differently now with NW SCP:SL 14 release

**Pull request closed without merging**